scripts/ioc_deploy.py

```diff
@@ -700,10 +718,40 @@ def _clone(
     return subprocess.run(cmd, **kwds)
 
 
+def rearrange_sys_argv_for_subcommands():
+    """
+    Small trick to help argparse deal with my optional subcommand.
+
+    This will make argv like this:
+    ioc-deploy -p some_path perms ro
+    be interpretted the same as:
+    ioc-deploy perms ro -p some_path
+
+    Otherwise, the first example here is interpretted as if -p was never passed,
+    which could be confusing.
```

Comment:
This is truly an extra mile QOL addition, but I can't argue with the results.  I think normally sub-commands just don't work if you don't specify it as the first argument, and I'd have left it at that haha